sklearn/ensemble/_hist_gradient_boosting/binning.py

```diff
@@ -6,6 +6,7 @@
 approximately the same number of samples.
 """
 # Author: Nicolas Hug
+import concurrent.futures
```

Comment:
> Is there a reason, we never use a ThreadPoolExecutor?

I think one is used by `joblib` when [`loky`](https://github.com/joblib/loky) as a back-end.

Reply:
But this here uses `ThreadPoolExecutor`, not `ProcessPoolExecutor`.

Reply:
I see.

I think @ogrisel is much more knowledgeable than me to make an educated decision.

Reply:
This pattern is interesting ; I wonder whether we could abstract it and reuse it in some places.

Reply:
I confirm that loky is only useful for process-based parallelism. For thread-based parallelism, `ThreadPoolExecutor` is perfectly fine.